bindings/jni/src/dimensionality.cpp

```diff
@@ -22,9 +22,18 @@ JNIEXPORT jlongArray JNICALL Java_io_shapelets_khiva_Dimensionality_paa(JNIEnv *
     jlong raw_pointer = 0;
     af_array af_p = (af_array)raw_pointer;
 
-    af_retain_array(&arr, var.get());
-    af_retain_array(&af_p, khiva::dimensionality::PAA(var, bins).get());
+    char const *className = "java/lang/RuntimeException";
+    jclass exClass = env->FindClass(className);
 
+    af_retain_array(&arr, var.get());
+    try {
+        af_retain_array(&af_p, khiva::dimensionality::PAA(var, bins).get());
+    } catch (std::invalid_argument &ia) {
+        char const *message =
+            "PAA invalid argument. The number of important points should be a factor of the total number of points. "
```

Comment:
"The number of important points should be a factor of the total number of points." should come from the exception message